src/mam4xx/wet_dep.hpp

```diff
@@ -0,0 +1,178 @@
+// mam4xx: Copyright (c) 2022,
+// Battelle Memorial Institute and
+// National Technology & Engineering Solutions of Sandia, LLC (NTESS)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef MAM4XX_WET_DEPOSITION_HPP
+#define MAM4XX_WET_DEPOSITION_HPP
+
+#include <haero/atmosphere.hpp>
+#include <haero/constants.hpp>
+#include <haero/math.hpp>
+#include <mam4xx/aero_config.hpp>
+
+// Based on e3sm_mam4_refactor/components/eam/src/chemistry/aerosol/wetdep.F90
+namespace mam4 {
+
+namespace wetdep {
+
+/**
+ * @brief Calculate local precipitation generation rate (kg/m2/s)
+ *        from source (condensation) and sink (evaporation) terms.
+ *
+ * @param[in] pdel Pressure difference across layers [Pa].
+ * @param[in] source_term Precipitation source term rate (condensation)
+ * [kg/kg/s].
+ * @param[in] sink_term Precipitation sink term rate (evaporation) [kg/kg/s].
+ * @param[in] atm Atmosphere object (used for number of levels).
+ *
+ * @param[out] lprec Local production rate of precipitation [kg/m2/s].
+ *
+ * @pre pdel, source_term, sink_term and lprec are all an array
+ *      of size pver == atm.num_levels().
+ *
+ * @pre In F90, ncol == 1 as we only operate over one column at a time
+ *      as outer loops will iterate over columns, so we drop ncol as input.
+ * @pre In F90 code, pcols is the number of columns in the mesh.
+ *      Since we are only operating over one column, pcols == 1.
+ * @pre In F90 code, ncol == pcols. Since ncol == 1, pcols == 1.
+ *
+ * @pre atm is initialized correctly and has the correct number of levels.
+ */
+KOKKOS_INLINE_FUNCTION
+void local_precip_production(/* cont int ncol, */ const Real *pdel,
+                             const Real *source_term, const Real *sink_term,
+                             Real *lprec, const Atmosphere &atm) {
+  const int pver = atm.num_levels();
+  for (int i = 0; i < pver; i++) {
+    lprec[i] = (pdel[i] / Constants::gravity) * (source_term[i] - sink_term[i]);
+  }
+}
+
+// clang-format off
```

Comment:
It doesn't matter but using the `k` index instead of `i` is more natural for me when thinking about the vertical levels. Maybe it is just me :-)

Reply:
@jaelynlitz mentioned something about also potentially just passing in the number of levels into all of these functions instead of the entire atmosphere object - I will change these to k and also fix this in the same fix if others agree